braindecode/datasets/sleep_physionet.py

```diff
@@ -99,3 +106,151 @@ def _load_raw(raw_fname, ann_fname, preload, load_eeg_only=True,
+    rng = np.random.RandomState(random_state)
+    tf_list_len = len(transform_list)
+    len_aug_dataset = len(train_dataset) * tf_list_len
+    subset_sample = rng.choice(
+        range(int(len_aug_dataset / tf_list_len)),
+        size=int(sample_size *
+                 len_aug_dataset /
+                 tf_list_len),
+        replace=False)
+    subset_aug_sample = np.array([(np.arange(i * tf_list_len, i * tf_list_len +
+                                             tf_list_len))
+                                  for i in subset_sample]).flatten()
+    subset_aug_labels = np.array([(np.full(tf_list_len, train_dataset[i][1]))
+                                  for i in subset_sample]).flatten()
+
+    # train_subset = Subset(
+    #     dataset=train_dataset,
+    #     indices=subset_aug_sample)
+    return subset_aug_sample, subset_aug_labels
+
+    # TODO Trouver autre solution subset.
+    # TODO mne-tools/mne-features
+
+
+def build_epoch(subjects, recording, crop_wake_mins, preprocessing,
+                train=True):
+    dataset = SleepPhysionet(subject_ids=subjects,
+                             recording_ids=recording,
+                             crop_wake_mins=crop_wake_mins)
+
+    if preprocessing:
+        preprocessors = []
+        if "microvolt_scaling" in preprocessing:
+            preprocessors.append(NumpyPreproc(fn=lambda x: x * 1e6))
+        if "filtering" in preprocessing:
+            high_cut_hz = 30
+            preprocessors.append(
+                MNEPreproc(fn='filter', l_freq=None, h_freq=high_cut_hz)
+            )
+
+        # Transform the data
+        preprocess(dataset, preprocessors)
+    mapping = {  # We merge stages 3 and 4 following AASM standards.
+        'Sleep stage W': 0,
+        'Sleep stage 1': 1,
+        'Sleep stage 2': 2,
+        'Sleep stage 3': 3,
+        'Sleep stage 4': 3,
+        'Sleep stage R': 4
+    }
+
+    window_size_s = 30
+    sfreq = 100
+    window_size_samples = window_size_s * sfreq
+
+    windows_dataset = create_windows_from_events(
+        dataset, trial_start_offset_samples=0,
+        trial_stop_offset_samples=0,
+        window_size_samples=window_size_samples,
+        window_stride_samples=window_size_samples, preload=True,
+        mapping=mapping)
+
+    return windows_dataset
+
+
+def tinying_dataset(concat_dataset, subset_dict):
+
+    def take_dataset_subset(windows_dataset, indice_list):
+        windows_dataset.windows = windows_dataset.windows[tuple(
+            indice_list)]
+        windows_dataset.y = windows_dataset.y[indice_list]
+        windows_dataset.crop_inds = windows_dataset.crop_inds[indice_list]
+        return(windows_dataset)
+
+    for i in subset_dict.keys():
+        concat_dataset.datasets[i] = take_dataset_subset(
+            concat_dataset.datasets[i], subset_dict[i])
+    concat_dataset.cumulative_sizes = concat_dataset.cumsum(
+        concat_dataset.datasets)
+    return(concat_dataset)
```

Comment:
I don't feel this is library code. It's utils for your own experiments on sleep data. I would not be part of the core library